pyvista/utilities/fileio.py

```diff
@@ -8,6 +8,10 @@
 
 import imageio
 
+import numpy
+
+import meshio
```

Comment:
I am a bit puzzled you add this as a hard dependency. I would go with nested imports.
Also I would recommend to have a travis build that runs with only hard dependencies (numpy + vtk).

my 2c

Reply:
It's a pure python module, so there's not much harm done by adding `meshio` as a dependency, but it's something to consider as I too don't like having a ton of dependencies.  However, I think it's worth the small additional install time for the user to be able to seamlessly use `meshio` within `pyvista`.

Reply:
I'm in agreement with @akaszynski here - `meshio` is lightweight enough and the benefits of the supported file types it will add make it worth-while. 

More supported file types ▶️  more interest ▶️ more users ▶️ more contributors ▶️ better/more stable PyVista

Reply:
I am not saying you should not support it I am suggesting you make it a nested import. In MNE for example we won't need this. Any dependency will have a cost for you in the future. For example you need to do a debian package, then you'll have to package 2 libraries separately. If meshio looses is core dev you'll have to maintain it or find an alternative etc. I don't have the authority here to make this call, but really think twice before adding deps. FYI mne and scikit-learn have as minimal deps numpy and scipy. PyVista is really for me a thin layer of pure python code on top of vtk so vtk+numpy should be all I need. my 2c

Reply:
> For example you need to do a debian package, then you'll have to package 2 libraries separately.

https://tracker.debian.org/pkg/python-meshio

Reply:
yes I saw it's packaged. again feel free to ignore my comment if you consider it inappropriate.

Reply:
That's cool that it's a debian package!

In many ways, I'm with you @agramfort.  I also don't have plans (at the moment) for using `meshio`, but if it can read in some ANSYS files, that might be something I look at in the future.  A nested import with an optional dependency would work just fine for me.  However, considering how huge `numpy` and `vtk` are, I don't think we should worry about a pure python package.  Correct me if I'm wrong, but I don't see where a debian package would be used for a pure python install if you're installing it using `pip`, and packaging it with conda should also take care of the dependency issues.

I agree that adding another dependency complicates things, and this is exactly why the dependencies for writing movies were left as an optional dependency with a nested import.  Installing it for me requires a *.deb package, and Windows requires extra work as well, and it's already tough to install VTK for some.

Since we should be mostly concerned with balancing the install/use experience, I see no added cost including a lightweight pure python dependency as this saves the user from going back through and running `pip install meshio` after they get a `ModuleNotFoundError`.  We already install `scooby` and `appdirs` which are optional for most plotting functions, but the extra step to install them when the user needs to print out a system report or download an example file wouldn't seem polished for me.

Reply:
In general, I'm in favour of [*"Always implement things when you actually need them, never when you just foresee that you need them."*](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).